File: app/src/main/java/com/example/barcode/MainActivity.java
package com.example.barcode;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private Button btnScanBarcode;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnScanBarcode = findViewById(R.id.btnScanBarcode);

        btnScanBarcode.setOnClickListener(view -> {
            Intent intent = new Intent(MainActivity.this, ScannedBarcodeActivity.class);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/barcode/MainActivity.kt
package com.example.barcode

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.barcode.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Initialize View Binding
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Set click listener using binding
        binding.btnScanBarcode.setOnClickListener {
            val intent = Intent(this, ScannedBarcodeActivity::class.java)
            startActivity(intent)
        }
    }
}
